src/main/java/org/spongepowered/api/GameRegistry.java

```diff
@@ -92,190 +48,13 @@
 public interface GameRegistry {
 
     /**
-     * Attempts to retrieve the specific type of {@link CatalogType} based on
-     * the string id given.
+     * Gets an instance of the provided {@link Provider} class.
      *
-     * <p>Some types may not be available for various reasons including but not
-     * restricted to: mods adding custom types, plugins providing custom types,
-     * game version changes.</p>
-     *
-     * @param typeClass The class of the type of {@link CatalogType}
-     * @param id The string id of the catalog type
-     * @param <T> The type of catalog type
-     * @return The found catalog type, if available
-     */
-    <T extends CatalogType> Optional<T> getType(Class<T> typeClass, String id);
-
-    /**
-     * Gets a collection of all available found specific types of
-     * {@link CatalogType} requested.
-     *
-     * <p>The presented {@link CatalogType}s may not exist in default catalogs
-     * due to various reasons including but not restricted to: mods, plugins,
-     * game changes.</p>
-     *
-     * @param typeClass The class of {@link CatalogType}
-     * @param <T> The type of {@link CatalogType}
-     * @return A collection of all known types of the requested catalog type
-     */
-    <T extends CatalogType> Collection<T> getAllOf(Class<T> typeClass);
```

Comment:
But it woudn't make any sense to put there. It's not that much longer and I think a lot of our API isn't immediately obvious which is why we use the javadocs and docs site to convey the needed information. This could easily be mentioned in the CatalogedBy annotation or other parts of the docs.